docs/typescript/types.md

```diff
@@ -101,6 +101,67 @@ When seeing this header, Deno would attempt to retrieve
 `https://example.com/coolLib.d.ts` and use that when type checking the original
 module.
 
+### Using ambient or global types
+
+Overall it is better to use module/UMD type definitions with Deno, where a
+module expressly imports the types it depends upon. Modular type definitions can
+express
+[augmentation of the global scope](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/global-modifying-module-d-ts.html)
+via the `declare global` in the type definition. For example:
```

Comment:
Hm, I don't agree that modules with side-effects on the whole program's type-checking are better to use than the introduced `"types"` config. They're very oxymoronic. Global augmentations are better represented at the CLI/config level.

Reply:
I guess we have to agree to disagree on this. Code expressing it dependencies/expectations as code makes the code more portable without requiring magical configuration.

Reply:
Global augmentations in modules are implicit dependencies. Exemplary usage of these is having the single entrypoint module in your project import the augmenter and leaving the rest with this implicit dependency. This is not more portable in the big picture. Needing to specify them at application-level (since they have an application-wide effect) would have the dual effect of making them explicit and discouraging them altogether. The explicitness is in the form of publicly documenting a list of `d.ts` URLs alongside your project which consumers cannot avoid mentally parsing and referencing at the root of their application.

But we can't remove the language feature, this is just a matter of doc priority, so agree to disagree.